controllers/action.py

```diff
@@ -359,6 +388,52 @@ def handle_resize_request():
             pass
 
 
+def handle_public_ip_request():
+    action = request.vars.action
+    if action == 'approve':
+        return __attach_vm_public_ip()
+    elif action == 'reject':
+        return __reject_public_ip()
+
+def __reject_public_ip():
+    db(db.floating_ip_requests.id == request.vars.id).delete()
+    return jsonify(action='reject')
+
+def __attach_vm_public_ip():
+    try:
+         auth = b64encode(dumps(dict(u=session.username, p=session.password)))
+         auth = Storage(loads(b64decode(auth)))
+         req = db(db.floating_ip_requests.id==request.vars.id).select().first()
+         conn = Baadal.Connection(_authurl, _tenant, session.username,
+                                  session.password)
+         vm = conn.find_baadal_vm(id=req.vmid)
+         vm_name = vm.name
+         vm.attach_floating_ip()
+         db(db.floating_ip_requests.id == request.vars.id).delete()
```

Comment:
Keeping old requests in the table can be beneficial for history and audit. It will be better to update the request instead of deleting it.